ocean_provider/validation/RBAC.py

```diff
@@ -27,10 +26,12 @@ def __init__(
             raise RequestNotFound("Request name is not valid!")
         self.action = action_mapping[request_name]
         self.provider_address = get_provider_wallet().address
-        self.credentials = {
-            "type": "address",
-            "address": self.provider_address,
-        }
+        address = (
+            self.request["consumerAddress"]
+            if "consumerAddress" in self.request and self.request["consumerAddress"]
```

Comment:
you can use self.request.get("consumerAddress") to avoid the checks, e.g. in one line:

`address = self.request.get("consumerAddress", self.request["publisherAddress"])`

Reply:
It looks cleaner now 👍🏻 
I adjusted your suggestion to this form `address = self.request.get("consumerAddress", self.request.get("publisherAddress"))`, because `publisherAddress` key is present just in the encrypt endpoint.